x-pack/plugins/ml/public/application/datavisualizer/index_based/components/field_data_row/action_menu/lens_utils.ts

```diff
@@ -0,0 +1,277 @@
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import type { LayerConfig } from '../../../../../../../../lens/public/xy_visualization/types';
+import type { CombinedQuery } from '../../../common';
+
+interface ColumnsAndLayer {
+  columns: Record<string, IndexPatternColumn>;
+  layer: LayerConfig;
+}
+
+export function getNumberSettings(item: FieldVisConfig, defaultIndexPattern: IndexPattern) {
+  // if index has no timestamp field
+  if (defaultIndexPattern.timeFieldName === undefined) {
+    const columns: Record<string, IndexPatternColumn> = {
+      col1: {
+        label: item.fieldName!,
+        dataType: 'number',
+        isBucketed: true,
+        operationType: 'range',
+        params: {
+          type: 'histogram',
+          maxBars: 'auto',
+          ranges: [],
+        },
+        sourceField: item.fieldName!,
+      },
+      col2: {
+        label: 'Count',
+        dataType: 'number',
+        isBucketed: false,
+        sourceField: 'Records',
+        operationType: 'count',
+      },
+    };
+
+    const layer: LayerConfig = {
+      accessors: ['col2'],
+      layerId: 'layer1',
+      seriesType: 'bar',
+      xAccessor: 'col1',
+    };
+    return { columns, layer };
+  }
+
+  const columns: Record<string, IndexPatternColumn> = {
+    col2: {
+      dataType: 'number',
+      isBucketed: false,
+      label: `Average of ${item.fieldName}`,
+      operationType: 'avg',
+      sourceField: item.fieldName!,
+    },
+    col1: {
+      dataType: 'date',
+      isBucketed: true,
+      label: defaultIndexPattern.timeFieldName!,
+      operationType: 'date_histogram',
+      params: { interval: 'auto' },
+      scale: 'interval',
+      sourceField: defaultIndexPattern.timeFieldName!,
+    },
+  };
+
+  const layer: LayerConfig = {
+    accessors: ['col2'],
+    layerId: 'layer1',
+    seriesType: 'line',
+    xAccessor: 'col1',
+  };
+
+  return { columns, layer };
+}
+export function getDateSettings(item: FieldVisConfig, defaultIndexPattern: IndexPattern) {
+  const columns: Record<string, IndexPatternColumn> = {
+    col2: {
+      dataType: 'number',
+      isBucketed: false,
+      label: 'Count of records',
+      operationType: 'count',
+      scale: 'ratio',
+      sourceField: 'Records',
+    },
+    col1: {
+      dataType: 'date',
+      isBucketed: true,
+      label: defaultIndexPattern.timeFieldName!,
+      operationType: 'date_histogram',
+      params: { interval: 'auto' },
+      scale: 'interval',
+      sourceField: defaultIndexPattern.timeFieldName!,
+    },
+  };
+  const layer: LayerConfig = {
+    accessors: ['col2'],
+    layerId: 'layer1',
+    seriesType: 'line',
+    xAccessor: 'col1',
+  };
+
+  return { columns, layer };
+}
+
+export function getKeywordSettings(item: FieldVisConfig) {
+  const columns: Record<string, IndexPatternColumn> = {
+    col1: {
+      label: 'Top value of category',
```

Comment:
does this string require translation?

Reply:
Updated here https://github.com/elastic/kibana/pull/89471/commits/a84a88526266049f3ded62fc4cbd666627b9132e